pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -54,230 +56,170 @@ func (e transientError) Error() string {
-		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
-			if !errors.IsNotFound(err) {
-				glog.V(2).Infof("Error while trying to get image stream tag %q: %v", istag, err)
-				return details, existingDeployments, latestDeploymentExists, nil
+// reconcileDeployments reconciles existing deployment replica counts which
+// could have diverged outside the deployment process (e.g. due to auto or
+// manual scaling, or partial deployments). The active deployment is the last
+// successful deployment, not necessarily the latest in terms of the config
+// version. The active deployment replica count should follow the config, and
+// all other deployments should be scaled to zero.
+//
+// Previously, scaling behavior was that the config replica count was used
+// only for initial deployments and the active deployment had to be scaled up
+// directly. To continue supporting that old behavior we must detect when the
+// deployment has been directly manipulated, and if so, preserve the directly
+// updated value and sync the config with the deployment.
+func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *kapi.ReplicationControllerList, config *deployapi.DeploymentConfig) error {
+	activeDeployment := deployutil.ActiveDeployment(config, existingDeployments)
+	_, latestDeployment := deployutil.LatestDeploymentInfo(config, existingDeployments)
+	activeDeploymentIsLatest := activeDeployment.Name == latestDeployment.Name
+	lastActiveReplicas, hasLastActiveReplicas := deployutil.DeploymentReplicas(activeDeployment)
+
+	// Compute the replica count for the active deployment (even if the active
+	// deployment doesn't exist). The active replica count is the value that
+	// should be assigned to the config, to allow the replica propagation to
+	// flow downward from the config.
+	//
+	// This takes into account resources predating the propagation behavior
+	// change, as well as external modifications to the deployments (e.g.
+	// scalers).
+	activeReplicas := config.Template.ControllerTemplate.Replicas
+	if activeDeployment != nil {
+		if activeDeploymentIsLatest {
+			if !hasLastActiveReplicas || lastActiveReplicas != activeDeployment.Spec.Replicas {
+				activeReplicas = activeDeployment.Spec.Replicas
 			}
-			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
-			// (a build configuration output points to it so it's going to be populated at some point in the
-			// future)
-			details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream tag %q does not exist.", istag, istag)
-			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
-			if err != nil {
-				glog.V(2).Infof("Error while trying to list build configs: %v", err)
-				return details, existingDeployments, latestDeploymentExists, nil
+		} else {
+			if activeDeployment.Spec.Replicas > 0 {
```

Comment:
> If somehow we don't have access to that desired annotation, then we have to do our best: if the active/old deployment is scaled >0, we assume that the fallback scaleup occured already based on the failed desired replicas

...and we assume that the fallback scaled-up active deployment's replica count should replace the DC's replica count?  Shouldn't we also check annotations here?